components/script/dom/idbrequest.rs

```diff
@@ -157,15 +166,29 @@ impl RequestListener {
                 .fire(request.upcast(), CanGc::note());
             transaction.set_active_flag(false);
         } else {
-            request.set_result(answer.handle());
-            Self::handle_async_request_error(request, &global);
+            // FIXME:(arihant2math) dispatch correct error
+            // Substep 2
+            Self::handle_async_request_error(&global, cx, request, Error::Data);
```

Comment:
Is it possible that tests are still failing because we're using an incorrect error here?